### DISCUSSION TITLE
How are events started/terminated?

### DISCUSSION THREAD

**Discussion by hilikstein** [blakeblackshear/frigate#2771] [Ask A Question]
I've started playing with it - looks great! already killed my deepstack instance as I was able to get better results after a couple of hours of integration.
What I am still strugling with is to understand the flow of events - I've seen instances where I received 3 notifications is a row for event start for the same movement - with different event ID for each of them.
In other cases, a single movement was broken to several events, with split seconds between them.

What I'm trying to understand is how does the system decide to terminate an event and start a new one, and if there's a way to control some sort of inactivity time for it to consider end of event?

**Comment by blakeblackshear** [comment]
The event ends if the tracked object goes missing for more frames than the configured `max_disappeared` value. By default, it is 5x the frame rate. Frames where the object is stationary do not count. If that value is increased too high, the next object may be assumed to be the missing object and you would get 1 long event rather than 2 separate events.

Also, note that the object tracking is fairly simplistic at this time. If two people walk past each other, Frigate may get confused and swap their event IDs.

**Reply by hilikstein** [reply]
Thanks! Seems like you've answered my question, yet now I am even more confused :)
If I understand your correctly, if my camera sees 2 persons and a dog simultaneously - it will track three concurrent events?
Suppose I have configured images over mqtt through the relevant topic - how can I tell the two different persons apart and associate them to the relevant event IDs?
Is there any sort of a aggregating entity? support I consider the 2 persons + dog as one incident - how can I group them? through motion on till motion off ?

**Reply by blakeblackshear** [reply]
Yes. Each object is a unique event. If you are just using the binary motion sensors for notifications, I recommend getting a snapshot of the camera entity at the time of trigger to get a current image. If you need/want to get granular about notifications, you will want to use the `events` mqtt topic to get information on the current tracked objects and get the specific id of the event. It has event ids for each and publishes `new\update\end` event types. There are some aggregating entities already in home assistant that leverage the mqtt topics of object type for an entire camera or zone.
